src/handler.js

```diff
@@ -99,3 +100,34 @@ exports.onPreResponse = function (request, reply) {
 
   return reply.continue
 }
+
+/**
+ * Function to be used to handle the 'onPreHandler' Hapi server extension.
+ * This validates the error is a FSPIOPError and that it contains a valid error code
+ * format as per section 7.6 of "API Definition v1.0.docx".
+ *
+ * @param request the http request
+ * @param reply
+ * @returns {boolean|reply.continue|continue|((key?: IDBValidKey) => void)}
+ */
+exports.onPreHandler = function (request, reply) {
+  const response = request.response
+  let incomingErrorCode
+  if (response instanceof ErrorFactory.FSPIOPError || response.isBoom) {
+    try {
+      incomingErrorCode = response.toApiErrorObject.ErrorInformation.errorCode
+      if (!ErrorFactory.validateFSPIOPErrorCode(incomingErrorCode)) {
+        // TODO: validateFSPIOPErrorCode throws an exception if code is not valid at which point
+        //  we nee to call response.takeover()
+        // response.takeover(ErrorFactory.createFSPIOPError(Errors.VALIDATION_ERROR, `The incoming error code: ${incomingErrorCode} is not a valid mojaloop specification error code`))
+        return reply.response(ErrorFactory.createFSPIOPError(Errors.VALIDATION_ERROR, `The incoming error code: ${incomingErrorCode} is not a valid mojaloop specification error code`)).code(400)
```

Comment:
Two things here: 
1. I believe you need to add the `.takeover()` at the end of the calling stack
2. You will need to ensure that you `toApiErrorObject` the error is also called on the FSPIOPError assuming that the prehandler is skipped (which would have done this for you). Please verify if this is required.

Reply:
Addressed with last commit